spec/requests/api/v1/users_spec.rb

```diff
@@ -105,11 +98,64 @@
     context "when using a lead provider token" do
       let(:token) { LeadProviderApiToken.create_with_random_token!(lead_provider: create(:lead_provider)) }
 
-      it "returns 401 for invalid bearer token" do
+      it "returns 403 for invalid bearer token" do
         default_headers[:Authorization] = bearer_token
         get "/api/v1/users"
         expect(response.status).to eq 403
       end
     end
   end
+
+  describe "#create" do
+    context "when authorized" do
+      before do
+        default_headers[:Authorization] = bearer_token
+        default_headers["Content-Type"] = "application/vnd.api+json"
+      end
+
+      let(:json) do
+        {
+          data: {
+            type: "user", # is this singular or plural??? # spec is agnostic # i chose singular
+            attributes: {
+              full_name: "John Doe",
+              email: "john.doe@example.com",
+            },
+          },
+        }.to_json
```

Comment:
That's what we need to send to the api?

I don't really understand why we would not go with something like
```
{
  full_name: "John Doe",
  email: "john.doe@example.com"
}
```
Everything else seems to be, well, a bit excessive. In particular, why do we send a `user` type to an api with `users` in the path - will the api not know that it expects user data? Will it do something else if we send it `school` type? 

`participant-declarations` api, at least judging by their documentation, goes with the minimalist approach. Their swagger example is
```
{
  "participant_id": "db3a7848-7308-4879-942a-c4a70ced400a",
  "declaration_type": "started",
  "declaration_date": "2021-05-31T02:21:32Z"
}
```
Thoughts?

Reply:
my understanding is we are conforming to the jsonapi spec. which documents here https://jsonapi.org/format/#crud-creating for the shape of what creating a resource is. my change should conform and the change you are suggesting would make it not conform. or are we not conforming to jsonapi spec?

Reply:
It's our internal API, we can choose to make it whatever we want it to be. Is conforming to the `jsonapi` spec making life easier for us? In this PR it probably is not, cause you need to do
```
        user = User.find_or_create_by!(email: params[:data][:attributes][:email]) do |u|
          u.full_name = params[:data][:attributes][:full_name]
        end
```
to get the relevant data. And the users controller happily ignores the `type` parameter that you are sending its way, which makes me think the spec was written for something much more generic than our use case - say an endpoint that needs to handle creating many, many types of objects.

If we really think that conforming to that spec is important, then we should probably install some gems that do it for us, like what `teacher-training-api` are doing. And probably make some changes to participant details api :P 

Something to discuss on dev catchup perhaps?

Reply:
Alright, so to recap, we will do our best to conform to the jsonapi spec. It means there will be some unwieldiness, like here, but hopefully the consistency will be worth it.